ruby/ql/lib/codeql/ruby/AST.qll

```diff
@@ -166,3 +167,20 @@ class RubyFile extends File {
   /** Gets the number of lines of comments in this file. */
   int getNumberOfLinesOfComments() { result = count(int line | this.line(line, true)) }
 }
+
+/**
+ * A successfully extracted file, that is, a file that was extracted and
+ * contains no extraction errors or warnings.
+ */
+class SuccessfullyExtractedFile extends RubyFile {
```

Comment:
If you don't want to restrict to Ruby source files only, you can do:
```suggestion
class SuccessfullyExtractedFile extends File {
```

Reply:
Thanks, done.